conversion.go

```diff
@@ -0,0 +1,152 @@
+package verkle
+
+import (
+	"bytes"
+	"fmt"
+	"runtime"
+	"sort"
+	"sync"
+)
+
+// BatchNewLeafNodeData is a struct that contains the data needed to create a new leaf node.
+type BatchNewLeafNodeData struct {
+	Stem   []byte
+	Values map[byte][]byte
+}
+
+// BatchNewLeafNode creates a new leaf node from the given data. It optimizes LeafNode creation
+// by batching expensive cryptography operations. It returns the LeafNodes sorted by stem.
+func BatchNewLeafNode(nodesValues []BatchNewLeafNodeData) []LeafNode {
```

Comment:
I'm using here something I did for the full tree conversion, but added parallelization at this layer. This wasn't needed in the tree conversion since, in that case we parallelized "at the client level" by working in subtrees.

But the idea is the same/similar.